libs/blog-bff/shared/util-wp/src/lib/wp-client.ts

```diff
@@ -16,14 +16,26 @@ export class WPRestClient {
 
   get<T>(url: string, params: Record<string, string>): Promise<WPRespone<T>> {
     const searchParams = new URLSearchParams(params);
-    return this.request(`${url}?${searchParams.toString()}`, { method: 'GET' });
+    return this.request(`/wp-json/wp/v2/${url}?${searchParams.toString()}`, {
+      method: 'GET',
+    });
+  }
+
+  postEmail<T>(body: T): Promise<WPRespone<T>> {
+    const form: FormData = new FormData();
+    form.append('email', body['email']);
+
+    return this.request(`/wp-json/newsletter/v2/subscribers`, {
+      body: form,
+      method: 'POST',
+    });
   }
 
   private async request<T>(
     url: string,
     { body, headers, method }: FetchConfig = {},
   ): Promise<WPRespone<T>> {
-    const request = await fetch(`${this.baseUrl}/wp-json/wp/v2/${url}`, {
+    const request = await fetch(`${this.baseUrl}${url}`, {
```

Comment:
keep it as it was. Instead add the new `namespace` argument to the constructor, which is gonna be `wp-json/wp/v2` or `wp-json/newsletter/v2`.
You can also add third optional argument to this called `options`, which is gonna have `namespace` with default value of `wp-json/wp/v2`

Reply:
Corrected